hs_core/models.py

```diff
@@ -1962,6 +1962,34 @@ def get_irods_storage(self):
         else:
             return IrodsStorage()
 
+    def set_quota_holder(self, setter, new_holder):
+        # set quota holder of the resource to new_holder who must be an owner
+        # setter is the requesting user to transfer quota holder and setter must also be an owner
+        if not setter.uaccess.owns_resource(self) or not new_holder.uaccess.owns_resource(self):
+            raise PermissionDenied("Only owners can set or be set as quota holder for the resource")
+        istorage = self.get_irods_storage()
+        istorage.setAVU(self.root_path, "quotaUserName", new_holder.username)
+
+    def get_quota_holder(self):
+        # get quota holder of the resource
+        # return User instance of the quota holder for the resource or None if it does not exist
+        istorage = self.get_irods_storage()
+        try:
+            uname = istorage.getAVU(self.root_path, "quotaUserName")
+        except SessionException:
+            # quotaUserName AVU does not exist, return None
+            return None
+
+        if uname:
+            ufilter = User.objects.filter(username=uname)
```

Comment:
@pkdash The simplified query you suggested was what I implemented initially, however, I have to change the implementation to it is now since there is an exception raised complaining NoneType object does not have first(), or something like that either when I deploy this work to alpha VM or when running access control tests, so I actually made this change at the last minute so that it returns None if that user does not exist. I don't recall exactly what causes this exception, but adding this safe guide fixed that, so I'd prefer to keep this safe guard there since I know at least for initial migration, we'll run into issues without this safe guard.